src/main/java/tools/map/making/JTextAreaOptionPane.java

```diff
@@ -24,7 +24,7 @@
   private final JFrame windowFrame = new JFrame();
   private final JButton okButton = new JButton();
   private final boolean logToSystemOut;
-  private final WeakReference<Window> parentComponentReference;
+  private final Window parentComponentReference;
```

Comment:
Might want to consider renaming to remove the `Reference` suffix in conjunction with the type change.